### PR TITLE
Spring WS docs and code cleanup.

### PR DESCRIPTION
**Pull request by ammachado**
# Description

This PR adds small changes to the Spring-WS component documentation, as well as it applies the `instanceof pattern` cleanup a few times.

# Target

- [x] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [x] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [x] I checked that each commit in the pull request has a meaningful subject line and body.
- [x] I have run `mvn clean install -DskipTests` locally from root folder and I have committed all auto-generated changes.

### CODE REVIEW

File: components/camel-spring-ws/src/main/java/org/apache/camel/component/spring/ws/SpringWebserviceComponent.java [hunk 1]
```diff
@@ -75,20 +75,19 @@ private void addConsumerConfiguration(
         if (type != null) {
             LOG.debug("Building Spring Web Services consumer of type {}", type);
             String lookupKey = getLookupKey(remaining, type);
+            configuration.setEndpointMappingType(type);
+            configuration.setEndpointMappingLookupKey(lookupKey);
             if (EndpointMappingType.BEANNAME.equals(type)) {
                 addEndpointDispatcherToConfiguration(configuration, lookupKey);
             } else {
                 addEndpointMappingToConfiguration(parameters, configuration);
+                if (type.equals(EndpointMappingType.XPATHRESULT)) {
```
Comment (davsclaus) [inline components/camel-spring-ws/src/main/java/org/apache/camel/component/spring/ws/SpringWebserviceComponent.java:84]: it seem this code is moved into this else block which was not the case before.

Comment (ammachado) [reply components/camel-spring-ws/src/main/java/org/apache/camel/component/spring/ws/SpringWebserviceComponent.java:84]: You are correct, but this block will only execute if the branch where it is now is executed anyway, so I moved them together.

**Comment by davsclaus** [comment]
modified:   components/camel-spring-ws/src/main/java/org/apache/camel/component/spring/ws/bean/WSACamelEndpointMapping.java
	modified:   components/camel-spring-ws/src/main/java/org/apache/camel/component/spring/ws/converter/StringSourceConverter.java

File: components/camel-spring-ws/src/main/java/org/apache/camel/component/spring/ws/SpringWebserviceComponent.java (hunk 1 above)
Comment (orpiske) [reply components/camel-spring-ws/src/main/java/org/apache/camel/component/spring/ws/SpringWebserviceComponent.java:84]: A bit of a nitpick, but ideally it should be on a separate commit, as it prevents us from cherry-picking the doc changes to 4.4.x if we would like to.

**Review by orpiske — COMMENTED** [review]
Those are great changes. 

IMHO it would be better to have them on separate commits, so we can cherry-pick changes if we want to.

**Comment by davsclaus** [comment]
@ammachado there are 2 modified files. If you run mvn clean install in the folder then it would update those files.

**Comment by ammachado** [comment]
My local build has some failing tests on `camel-openapi-java`, that's why I missed these changes. I'll open a JIRA for them.

**Comment by ammachado** [comment]
I added a PR on Spring-WS to upgrade WSS4J as well. https://github.com/spring-projects/spring-ws/pull/1429 and match the version used in Camel.

**Review by davsclaus — APPROVED** [review]

**Pull request merged**